sdi/helper.go

```diff
@@ -0,0 +1,71 @@
+package sdi
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"mime"
+	"mime/multipart"
+	"strings"
+
+	resty "github.com/go-resty/resty/v2"
+)
+
+// ExtractXMLContent extracts the content inside specified XML tag.
+func ExtractXMLContent(data string, tag string) (string, error) {
+	openingTag := "<" + tag + ">"
+	closingTag := "</" + tag + ">"
+
+	openingTagIndex := strings.Index(data, openingTag)
+	if openingTagIndex == -1 {
+		return "", fmt.Errorf("opening tag <%s> not found", tag)
+	}
+
+	closingTagIndex := strings.Index(data, closingTag)
+	if closingTagIndex == -1 {
+		return "", fmt.Errorf("closing tag </%s> not found", tag)
+	}
+
+	closingTagIndex += len(closingTag)
+
+	extractedXML := data[openingTagIndex:closingTagIndex]
+	return extractedXML, nil
+}
+
+// ParseMultipartResponse parses a multipart HTTP response and deserializes the content into the provided structure
+func ParseMultipartResponse(resp *resty.Response, response interface{}) error {
```

Comment:
This probably doesn't need to be a public method either.

Reply:
Good point, this function can be private.